### ISSUE TITLE
Send SATS fails and lands on unknown route

### ISSUE THREAD

**Issue by pete-watters** [blockstack/blockstack-app#4635]
I was testing [this issue](https://github.com/leather-wallet/extension/issues/4603) and encountered a bug when trying to send `SATS`. 

I have `SATS` in one account and wanted to transfer them to another. I am not sure if this is supposed to work, any ideas @markmhendrickson ? 

It seems weird as 
- the route for `send/brc20/sats/error` page I land on isn't defined
- when I am doing a `Send` and choosing `SATS` it doesn't ask me where I want to send them

https://github.com/leather-wallet/extension/assets/2938440/f6b65e74-703f-4dde-b155-d28918d8c602

**Comment by markmhendrickson** [comment]
I'm getting the same but unsure the problem – will need checking the error logs, etc. to diagnose

The user should see a confirmation screen upon submission of the review screen

**Comment by markmhendrickson** [comment]
@pete-watters FYI I've moved this to the top of the backlog

**Comment by markmhendrickson** [comment]
I'm also seeing a related issue where my balance isn't recognized during sending. Prefer I file a separate issue or want to tackle jointly here?

<img width="403" alt="Screenshot 2023-12-08 at 07 36 28" src="https://github.com/leather-wallet/extension/assets/28991/c66c73c6-d31f-49b7-a52b-489c88e121d1">

<img width="421" alt="Screenshot 2023-12-08 at 07 36 19" src="https://github.com/leather-wallet/extension/assets/28991/e2c8a99e-a81e-431d-beb2-97ff58951acf">

**Comment by pete-watters** [comment]
> I'm also seeing a related issue where my balance isn't recognized during sending. Prefer I file a separate issue or want to tackle jointly here?
> 
> <img alt="Screenshot 2023-12-08 at 07 36 28" width="403" src="https://private-user-images.githubusercontent.com/28991/288985248-c66c73c6-d31f-49b7-a52b-489c88e121d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMjc2NzMsIm5iZiI6MTcwMjAyNzM3MywicGF0aCI6Ii8yODk5MS8yODg5ODUyNDgtYzY2YzczYzYtZDMxZi00OWI3LWE1MmItNDg5Yzg4ZTEyMWQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA5MjI1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyMDEyNGZhNGI4MTM3ZmUzOWNkN2FjYjBhNjM1ZjEzMGRiZDlmYTQzMzlmMDZiNjAxYTRjYWJiNWM4N2FjZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rMgdqJ4QWnYKx7sKgwybJ2eBn--pKU_wIcBye8PiY1k"> <img alt="Screenshot 2023-12-08 at 07 36 19" width="421" src="https://private-user-images.githubusercontent.com/28991/288985240-e2c8a99e-a81e-431d-beb2-97ff58951acf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMjc2NzMsIm5iZiI6MTcwMjAyNzM3MywicGF0aCI6Ii8yODk5MS8yODg5ODUyNDAtZTJjOGE5OWUtYTgxZS00MzFkLWJlYjItOTdmZjU4OTUxYWNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA5MjI1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlYWQ0NWZkYjM1MGYwY2JhZTA1OWRlNTQzOTlhZDg4MjFkYjFlMjIyYzFhOWI1Njg2ZmQyYjc0MDE2NjgxMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.51-AsyFjo77BAec_QgB-7qd6UwU9vSFTSeuSHLF0y8A">

It's OK to have them both fixed in this issue. Do you know if there was a design to go with this originally or else a ticket for how it's supposed to work?

I can try figure it out anyway. It seems it fails at creating the inscription which is the step before choosing a recipient?

**Comment by markmhendrickson** [comment]
@mica000 mind linking any of the BRC-20 materials we might have in Figma?

**Comment by markmhendrickson** [comment]
> It seems it fails at creating the inscription which is the step before choosing a recipient?

Sending BRC-20 is unique in that it's a 2-step process, in which during the send flow, the only thing that really happens is the creation of an inscription via the OrdinalsBot API. No recipient is indicated within the send flow at this time.

Then later, the user comes back to the home screen and sends the inscription via the normal flow, indicating the recipient at that point.

**Comment by markmhendrickson** [comment]
It's also entirely possible there's an underlying Ordinalsbot API error thrown here and we aren't catching it, so it's worth checking. And we can contact their team as needed to help diagnose.

**Comment by pete-watters** [comment]
I checked this out and after clicking `Create transfer inscription` we get a `400 Bad Request` error from `https://blockstream.info/api/tx`

It complains about: 
```
sendrawtransaction RPC error: {"code":-26,"message":"non-mandatory-script-verify-flag (Witness program hash mismatch)"}
```

Then we crash unhandled. 

This method fails in `bitcoin-client.ts`:  
```
  async broadcastTransaction(tx: string) {
    // TODO: refactor to use `axios`
    // https://github.com/leather-wallet/extension/issues/4521
    // eslint-disable-next-line no-restricted-globals
    return fetch(`${this.configuration.baseUrl}/tx`, {
      method: 'POST',
      body: tx,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    });
  }
```

We added this functionality in [this release](https://github.com/leather-wallet/extension/pull/3790)

I think it may have been broken in [#4316](https://github.com/leather-wallet/extension/pull/4316)) by these refactors of `useGenerateUnsignedNativeSegwitSingleRecipientTx` :
* [this commit](https://github.com/leather-wallet/extension/pull/4316/files#diff-233c7972dadd88db470555a9596499ae1ad4ce709cc9adaaaf2e2e90ef83dc09)
* [this commit](https://github.com/leather-wallet/extension/pull/4316/files#diff-d37d8e2c79294616b472400c8514ca6cc2e0f4a0205f9ae90e338493f066b6c8)

Going back further to commit `e9da65164273476ae4ba0c2de185dc07c7230d34` and it was working so it's something on our side. Any ideas here @kyranjamie ?

![Screenshot 2023-12-08 at 14 31 36](https://github.com/leather-wallet/extension/assets/2938440/cad05338-48ff-4c48-9da2-34380746d28d)

**Comment by kyranjamie** [comment]
Yep, looks like the refactor in the Ledger work broke this. [We generate the tx](https://github.com/leather-wallet/extension/blob/6fca3f2c2d345def83b758da62e647397daf738d/src/app/pages/send/send-crypto-asset-form/form/brc-20/brc-20-choose-fee.tsx#L73-L80), then try and broadcast it.

Previously this hook generated and signed the tx, which it no longer does. We need to sign the transaction with `useSignBitcoinTx`, finalise it, then broadcast it.

**Comment by pete-watters** [comment]
Thanks for the help here @kyranjamie . I think I have fixed it [here](https://github.com/leather-wallet/extension/pull/4665) as I submitted some things successfully. 

@markmhendrickson in my fixing of this I had to spend ~ `$70` on fees, sending a couple of transactions:
![Screenshot 2023-12-08 at 15 55 09](https://github.com/leather-wallet/extension/assets/2938440/c2d2225b-9d4e-4885-927c-e3bcb75800f3)

I saw what I think is another bug also where if I sign out of my wallet and then into our test wallet, my `Pending BRC-20 transfers` showed in the other wallet. This could be some weird caching issue. 


https://github.com/leather-wallet/extension/assets/2938440/42c4a13f-a738-4470-a73a-5c3e1e28c54c


This bug is fixed but I'm confused what the next step for moving these tokens is

1. I think we can have a better UX for where we show the Pending BTC-20 transfers - maybe it should be in `Activity`?
2. The `Pending` part is gone from my wallet now, as is the cost of the fees but I don't know what I'm supposed to do next. After some time should an Ordinal Inscription show up in my collectibles? If so, maybe we can add a placeholder entry there as a visual queue
